bgpd/bgp_vty.c

```diff
@@ -624,18 +624,12 @@ static int bgp_clear(struct vty *vty, struct bgp *bgp, afi_t afi, safi_t safi,
 
 			if (ret < 0)
 				bgp_clear_vty_error(vty, peer, afi, safi, ret);
-			else
-				found = true;
 		}
 
 		/* This is to apply read-only mode on this clear. */
 		if (stype == BGP_CLEAR_SOFT_NONE)
 			bgp->update_delay_over = 0;
 
-		if (!found)
```

Comment:
should similar cleanup needs to be done @ line 735 for the below error ?

if (!found)
			vty_out(vty,
				"%%BGP: No external %s peer is configured\n",
				afi_safi_print(afi, safi));

Otherwise, code changes looks good. Let me know and I will approve and merge this.

Reply:
@srimohans -> We only need to worry about bgp_soft_star_clear_in() calling for each afi/safi.  So we will not be in the same code path.